packages/tree-widget/src/components/TreeWidgetComponent.tsx

```diff
@@ -0,0 +1,196 @@
+  WidgetControl,
+  UiFramework,
+} from "@bentley/ui-framework";
+import { ModelsTreeComponent } from "./ModelsTree";
+import { CategoriesTreeComponent } from "./CategoriesTree";
+import { StoriesTree } from "./StoriesTree";
+import "./TreeWidgetComponent.scss";
+import { ThemedSelect, OptionType } from "@bentley/ui-core";
+import { useState } from "react";
+import { RealityDataWidget } from "./RealityDataWidget";
+import { TreeWidget } from "../TreeWidget";
+
+interface TreeWidgetComponentProps {
+  imodel: IModelConnection;
+  accessToken: AccessToken;
+  activeView?: Viewport;
+  enablePreloading?: boolean;
+  config?: {
+    modelsTree: {
+      enableElementsClassGrouping?: boolean;
+      selectionMode?: SelectionMode;
+      selectionPredicate?: ModelsTreeSelectionPredicate;
+    };
+    categoriesTree: {
+      enableElementsClassGrouping?: boolean;
+      allViewports?: boolean;
+    };
+    spatialTree: {
+      allViewports?: boolean;
+    };
+    storyTree: {
+      allViewports?: boolean;
+    };
+  };
+}
+
+function TreeWidgetComponent(props: TreeWidgetComponentProps) {
+  const { imodel, activeView, enablePreloading, accessToken } = props;
+  const modelsTreeProps = props.config?.modelsTree;
+  const categoriesTreeProps = props.config?.categoriesTree;
+  const spatialTreeProps = props.config?.spatialTree;
+  const storyTreeProps = props.config?.storyTree;
+
+  const modelsTreeComponent = React.useCallback(
+    () => (
+      <ModelsTreeComponent
+        iModel={imodel}
+        activeView={activeView}
+        enablePreloading={enablePreloading}
+        {...modelsTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, modelsTreeProps]
+  );
+
+  const categoriesTreeComponent = React.useCallback(
+    () => (
+      <CategoriesTreeComponent
+        iModel={imodel}
+        activeView={activeView}
+        enablePreloading={enablePreloading}
+        {...categoriesTreeProps}
+      />
+    ),
+    [imodel, activeView, enablePreloading, categoriesTreeProps]
+  );
+
+  const spatialContainmentComponent = React.useCallback(
+    () => (
+      <SpatialContainmentTree
+        iModel={imodel}
+        enablePreloading={enablePreloading}
+        {...spatialTreeProps}
+      />
+    ),
+    [imodel, enablePreloading, spatialTreeProps]
+  );
+
+  const storiesTreeComponent = React.useCallback(
+    () => (
+      <StoriesTree
+        iModel={imodel}
+        enablePreloading={enablePreloading}
+        {...storyTreeProps}
+      />
+    ),
+    [imodel, enablePreloading, storyTreeProps]
+  );
+
+  const realityDataComponent = React.useCallback(
+    () => <RealityDataWidget iModel={imodel} accessToken={accessToken} />,
+    [imodel, accessToken]
+  );
+
+  const [renderedElement, setRenderedElement] = useState<JSX.Element>(
+    modelsTreeComponent
+  );
+  const themedSelectOptions = [
+    {
+      label: UiFramework.translate("visibilityWidget.modeltree"),
+      value: "model-tree",
+    },
+
```

Comment:
few extra new lines in this array

Reply:
you mean add a few extra lines or remove some? I removed the new lines in the array

Reply:
also I can't reply to comment about the props, but I did elevate `allViewports` and `enableElementsClassGrouping` to a higher level. Not sure what you meant though with the first two sentences

Reply:
Weird, i cant seem to comment on that other one either. Thanks for taking care of that. Since you elevated those props, you can disregard the first 2 sentences as its not an issue anymore.